main.go

```diff
@@ -192,29 +233,90 @@ func main() {
 			data, err := json.Marshal(releaseDscInitialization)
 			if err != nil {
 				setupLog.Error(err, "failed to get DscInitialization custom resource data")
+				return err
 			}
 			err = client.Patch(context.TODO(), releaseDscInitialization, client2.RawPatch(types.ApplyPatchType, data),
 				client2.ForceOwnership, client2.FieldOwner("opendatahub-operator"))
 			if err != nil {
 				setupLog.Error(err, "failed to update DscInitialization custom resource")
+				return err
 			}
 		default:
 			setupLog.Error(err, "failed to create DscInitialization custom resource")
-			os.Exit(1)
+			return err
 		}
 	}
-	if err := mgr.AddHealthzCheck("healthz", healthz.Ping); err != nil {
-		setupLog.Error(err, "unable to set up health check")
-		os.Exit(1)
-	}
-	if err := mgr.AddReadyzCheck("readyz", healthz.Ping); err != nil {
-		setupLog.Error(err, "unable to set up ready check")
-		os.Exit(1)
+	return nil
+}
+
+func initializeDataScienceCluster(mgr manager.Manager) error {
+	// Create DSCInitialization CR if it's not present
+	fmt.Println("Entering dsc function")
+	client := mgr.GetClient()
+	platform, err := deploy.GetPlatform(client)
+	fmt.Println(err.Error())
+	if err != nil {
+		return fmt.Errorf("failed to get platform: %v", err)
 	}
+	fmt.Printf("platform is name: %v", platform)
 
-	setupLog.Info("starting manager")
-	if err := mgr.Start(ctrl.SetupSignalHandler()); err != nil {
-		setupLog.Error(err, "problem running manager")
-		os.Exit(1)
+	if platform != deploy.OpenDataHub && platform != deploy.ManagedRhods {
```

Comment:
so we want to create DSC (only kserve disabled) if it is selfmanaged?

Reply:
No I need to remove the deploy.OpenDataHub from the condition, but essentially return without creating DSC only when platform is Managed.

For Self-managed and ODH we do not create the Default DSC